parastell/nwl_utils.py

```diff
@@ -159,14 +161,27 @@ def flux_coords(vmec, wall_s, coords):
         theta_coords (array of float): poloidal angles of surface crossings
             (rad).
     """
-    phi_coords = np.arctan2(coords[:,1], coords[:,0])
-    theta_coords = []
-    
-    for pt, phi in zip(coords, phi_coords):
-        theta = find_coords(vmec, wall_s, phi, pt)
-        theta_coords.append(theta)
 
-    return phi_coords, theta_coords
+    phi_coords = np.arctan2(coords[:, 1], coords[:, 0])
+    chunk_size = math.ceil(len(phi_coords) / num_threads)
+    chunks = []
+    for i in range(num_threads):
+        chunk = list(
+            zip(
+                phi_coords[i * chunk_size : (i + 1) * chunk_size],
+                coords[i * chunk_size : (i + 1) * chunk_size],
+            )
+        )
+        chunks.append((plas_eq, wall_s, chunk))
+
+    with concurrent.futures.ProcessPoolExecutor(
+        max_workers=num_threads
+    ) as executor:
+        theta_coords = list(executor.map(find_coords, chunks))
+        theta_coords = [
+            theta_coord for chunk in theta_coords for theta_coord in chunk
+        ]
```

Comment:
Making sure I understand with new variable names
```suggestion
        theta_coord_chunks = list(executor.map(find_coords, chunks))
        theta_coords = [
            theta_coord for theta_coord_chunk in theta_coord_chunks for theta_coord in theta_coord_chunk
        ]
```